### PR TITLE
Convert 6 tests to use subtests #1520 (prefork, promise_async_await, response, roles, subprocess_ev, tls)

### PR DESCRIPTION
**Pull request by spazm**
### Summary
Convert 6 tests to use subtests (prefork, promise_async_await, response, roles, subprocess_ev, tls)

### Motivation
Requested in #1520 

### References
Requested in #1520 

Viewing the diff ignoring whitespace makes the changes a lot clearer.  `git diff -w` or add `&w=1` to the diffs url

Please feel free to edit or make any stylistic changes as you see fit.  Just looking to toss in a helping hand.

### CODE REVIEW

File: t/mojo/promise_async_await.t [hunk 1]
```diff
@@ -17,125 +17,136 @@ use Mojo::Promise;
...
 
-# async/await spec
-test_awaitable('Mojo::Promise conforms to Awaitable API', class => "Mojo::Promise", force => sub { shift->wait },);
+subtest "async/await spec" => sub {
+  test_awaitable('Mojo::Promise conforms to Awaitable API', class => "Mojo::Promise", force => sub { shift->wait },);
+};
 
-# Silence
-app->log->level('fatal');
+subtest "Silence" => sub {
```
Comment (kraih) [inline t/mojo/promise_async_await.t:24]: What does this even mean?

File: t/mojo/promise_async_await.t [hunk 2]
```diff
@@ -17,125 +17,136 @@ use Mojo::Promise;
...
+  };
+
+  helper defer_reject_p => sub {
+    my ($c, $msg) = @_;
+    my $promise = Mojo::Promise->new;
+    Mojo::IOLoop->next_tick(sub { $promise->reject($msg) });
+    return $promise;
+  };
+
+  get '/one' => {text => 'works!'};
```
Comment (kraih) [inline t/mojo/promise_async_await.t:41]: Why are these in a subtest?

**Comment by kraih** [comment]
These tests were super broken yesterday with syntax errors everywhere, like a bad script converted them. So please make sure to pay extra attention when reviewing.

File: t/mojo/prefork.t [hunk 3]
```diff
@@ -11,126 +11,143 @@ use Mojo::IOLoop::Server;
...
-$prefork->on(finish => sub { $graceful = pop });
-$prefork->run;
-is $prefork->ioloop->max_accepts, 500, 'right value';
-is scalar @spawn, 1, 'one worker spawned';
-is scalar @reap,  1, 'one worker reaped';
-ok !$graceful, 'server has been stopped immediately';
-is $tx->res->code, 200,          'right status';
-is $tx->res->body, 'works too!', 'right content';
+
+subtest "Manage and clean up PID file" => sub {
```
Comment (kraih) [inline t/mojo/prefork.t:17]: Double quotes are inconsistent.

**Review by kraih — CHANGES_REQUESTED** [review]
Some tests are fine and could be merged, but others have scoping issues and require more attention.

**Comment by kraih** [comment]
I'm sorry, but i've decided to close these PRs, because they contain too many converted tests at once. And we don't have the review capacity to re-review the whole thing after every scoping issue fix.

**Comment by spazm** [comment]
updated and rebased to remove the tests that didn't convert cleanly.  These four required minimal changes, as detailed in the updated PR summary.

**Pull request closed without merging**